commands/mail.js

```diff
@@ -181,24 +178,68 @@ module.exports = {
     request.method = "POST";
     request.url = "/v3/mail/send";
 
-    log.sys(`stringify request made by client: ${JSON.stringify(client.createRequest(request))}`);
+    const clientRequest = { ...client.createRequest(request) }; // Shallow copy
+    // remove every "Authorization"
+    unsetField(clientRequest, "Authorization");
+    log.sys(`stringify request made by client: ${JSON.stringify(clientRequest)}`);
 
     try {
-      await client.request(request);
-      log.good("Email with Sendgrid successfully sent");
+      const clientResponse = await client
+        .request(request)
+        .then(([response, body]) => {
+          // log.debug(`response.statusCode: ${response.statusCode} \n response.body: ${response.body} \n ${body}`);
+          log.debug(`response.statusCode: ${response.statusCode}`); // for success body is empty
+          return response;
+        })
+        .catch(err => {
+          log.debug(`code:${err.code}`);
+          log.debug(`message:${err.message}`);
+          log.debug(`response.headers:${JSON.stringify(err.response.headers)}`);
+          log.debug(`response.body:${JSON.stringify(err.response.body)}`);
+          return err;
+          // throw err; // We do not want to go to the catch (:262) we want to process this one (:258)
+        });
+      // https://docs.sendgrid.com/api-reference/how-to-use-the-sendgrid-v3-api/responses#status-codes
+      if (/2\d\d/g.test(clientResponse.statusCode)) {
+        log.good("Email with Sendgrid successfully sent");
+      } else {
+        log.err("Email with Sendgrid was NOT sent");
+        log.debug(`statusCode: ${clientResponse.code} \n message: ${clientResponse.message} \n ${clientResponse.toString()}`);
```

Comment:
Is any of this needed outside of debug mode such as the clientResponse.code so it can be looked up?

And then the detailed message and response can be in debug mode?

Reply:
@tlawrie  I will return the code, I've left above a link to the status codes that are returned and could be checked. Will adapt the code.